data-prepper-plugins/kafka-plugins/src/integrationTest/java/org/opensearch/dataprepper/plugins/kafka/source/ConfluentKafkaProducerConsumerWithSchemaRegistryIT.java

```diff
@@ -0,0 +1,376 @@
+        when(jsonTopicConfig.getConsumerMaxPollRecords()).thenReturn(100);
+        when(jsonTopicConfig.getMaxPollInterval()).thenReturn(Duration.ofSeconds(15));
+
+        avroTopicConfig = mock(TopicConfig.class);
+        avroTopicName = System.getProperty("tests.kafka.avro_topic_name");
+        when(avroTopicConfig.getName()).thenReturn(avroTopicName);
+        when(avroTopicConfig.getGroupId()).thenReturn("testGroupConf");
+        when(avroTopicConfig.getWorkers()).thenReturn(1);
+        when(avroTopicConfig.getSerdeFormat()).thenReturn(MessageFormat.PLAINTEXT);
+        when(avroTopicConfig.getAutoCommit()).thenReturn(false);
+        when(avroTopicConfig.getAutoOffsetReset()).thenReturn("earliest");
+        when(avroTopicConfig.getThreadWaitingTime()).thenReturn(Duration.ofSeconds(10));
+        when(avroTopicConfig.getSessionTimeOut()).thenReturn(Duration.ofSeconds(30));
+        when(avroTopicConfig.getHeartBeatInterval()).thenReturn(Duration.ofSeconds(5));
+        when(avroTopicConfig.getConsumerMaxPollRecords()).thenReturn(100);
+        when(avroTopicConfig.getMaxPollInterval()).thenReturn(Duration.ofSeconds(15));
+
+        encryptionConfig = mock(EncryptionConfig.class);
+
+        when(jsonSourceConfig.getAuthConfig()).thenReturn(authConfig);
+        when(jsonSourceConfig.getAcknowledgementsEnabled()).thenReturn(false);
+        when(jsonSourceConfig.getSchemaConfig()).thenReturn(schemaConfig);
+        when(jsonSourceConfig.getTopics()).thenReturn(List.of(jsonTopicConfig));
+        when(jsonSourceConfig.getBootstrapServers()).thenReturn(List.of(bootstrapServers));
+        when(jsonSourceConfig.getEncryptionConfig()).thenReturn(encryptionConfig);
+
+        when(avroSourceConfig.getAuthConfig()).thenReturn(authConfig);
+        when(avroSourceConfig.getAcknowledgementsEnabled()).thenReturn(false);
+        when(avroSourceConfig.getSchemaConfig()).thenReturn(schemaConfig);
+        when(avroSourceConfig.getTopics()).thenReturn(List.of(avroTopicConfig));
+        when(avroSourceConfig.getBootstrapServers()).thenReturn(List.of(bootstrapServers));
+        when(avroSourceConfig.getEncryptionConfig()).thenReturn(encryptionConfig);
+
+        when(encryptionConfig.getType()).thenReturn(EncryptionType.SSL);
+        when(plainTextAuthConfig.getUsername()).thenReturn(username);
+        when(plainTextAuthConfig.getPassword()).thenReturn(password);
+        when(schemaConfig.getBasicAuthCredentialsSource()).thenReturn("USER_INFO");
+        String[] userInfoParts = schemaRegistryUserInfo.split(":");
+        when(schemaConfig.getSchemaRegistryApiKey()).thenReturn(userInfoParts[0]);
+        when(schemaConfig.getSchemaRegistryApiSecret()).thenReturn(userInfoParts[1]);
+        when(schemaConfig.getRegistryURL()).thenReturn(schemaRegistryUrl);
+    }
+
+    @Test
+    public void KafkaJsonProducerConsumerTest() {
+        when(schemaConfig.getVersion()).thenReturn(2);
+        Thread producerThread = new Thread(() -> {
```

Comment:
As with my comment above - can we drop the threads? Typically tests should avoid using threads whenever possible.